app/models/asp/payment_request.rb

```diff
@@ -117,13 +117,25 @@ def active?
       !terminated?
     end
 
-    def eligible_for_auto_retry?
+    def eligible_for_incomplete_retry?
       return false unless in_state?(:incomplete)
 
       retryable_messages = RETRYABLE_INCOMPLETE_VALIDATION_TYPES.map do |r|
         I18n.t("activerecord.errors.models.asp/payment_request.attributes.ready_state_validation.#{r}")
       end
       last_transition.metadata["incomplete_reasons"]["ready_state_validation"].intersect?(retryable_messages)
     end
+
+    def eligible_for_rejected_and_unpaid_auto_retry?
+      if in_state?(:rejected)
+        message = last_transition.metadata["Motif rejet"]
```

Comment:
Au lieu d'aller chercher la metadata comme ca tu pourrais re-utiliser les helpers `rejected_reason` and `unpaid_reason`